### ISSUE TITLE
num2date for dates before the start of the Gregorian calendar

### ISSUE THREAD

**Issue by spencerkclark** [Unidata/netcdftime#28]
xref: pydata/xarray#1929

Using the current version of `netCDF4.num2date` the following works:
```python
>>> from netCDF4 import num2date
>>> import numpy as np
>>> num2date(np.arange(2), 'days since 1000-01-01', 'gregorian')
array([datetime.datetime(1000, 1, 1, 0, 0),
       datetime.datetime(1000, 1, 2, 0, 0)], dtype=object)
```
However, using the current `netcdftime.num2date`, it does not:
```python
>>> from netcdftime import num2date
>>> import numpy as np
>>> num2date(np.arange(2), 'days since 1000-01-01', 'gregorian')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "netcdftime/_netcdftime.pyx", line 283, in netcdftime._netcdftime.num2date
  File "netcdftime/_netcdftime.pyx", line 1165, in netcdftime._netcdftime.utime.num2date
  File "netcdftime/_netcdftime.pyx", line 673, in netcdftime._netcdftime.DateFromJulianDay
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```
This error came up in the Xarray test suite (not in a real-world use-case).  In my reading, it seems the Gregorian calendar does not start until October 15th, 1582, so should this be possible (as it is in `netCDF4`) or should it not be?  In other words, in this case is Xarray testing something that doesn't make sense?  How does (or should?) `netcdftime` handle this situation?

**Comment by jswhit** [comment]
Seems to be like this should not be allowed - the proleptic gregorian calendar should be used instead.  Don't know why it works with the netcdftime in netcdf4-python master.

**Comment by jswhit** [comment]
on second though, negative years should be allowed - a fixed for this (ported from netcdf4-python) is included in pull request #29.

**Comment by spencerkclark** [comment]
Awesome, thanks @jswhit.

**Comment by jhamman** [comment]
closed via #29
